server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -139,6 +149,41 @@ public String getApplicationName() {
     return applicationName;
   }
 
+  /**
+   * Get the ServiceLock for this server process. May return null if called before the lock is
+   * acquired.
+   *
+   * @return lock ServiceLock or null
+   */
+  public abstract ServiceLock getLock();
+
+  public void startServiceLockVerificationThread() {
+    final long interval =
+        getConfiguration().getTimeInMillis(Property.GENERAL_SERVER_LOCK_VERIFICATION_INTERVAL);
+    if (interval > 0) {
+      verificationThread = Threads.createThread("service-lock-verification-thread",
+          OptionalInt.of(Thread.NORM_PRIORITY + 1), () -> {
+            while (true && serverThread.isAlive()) {
+              ServiceLock lock = getLock();
+              try {
+                if (lock != null && !lock.verifyLockAtSource()) {
```

Comment:
Would be nice to add some trace logging for this thread that could be flipped on to see what its doing.  Maybe a trace log before and after checking the lock.

Reply:
Added trace logging in 601aff5.